File: voidr-playwright/page_objects/cart_page.py
from page_objects.base_page import BasePage

class CartPage(BasePage):
    def __init__(self, page):
        super().__init__(page)
        self.firstNameField = 'input[name="firstName"]'
        self.lastNameField = 'input[name="lastName"]'
        self.postalCodeField = 'input[name="postalCode"]'
        self.continueButton = 'input[type="submit"]'
        self.finishButton = 'button[name="finish"]'
        self.thankYouMessage = 'h2'
        self.checkOutError = 'h3'

    def checkout(self):
        self.page.locator('button[id="checkout"]').click()

    def fillCheckOutInformation(self, firstName, lastName, postalCode):
        self.page.fill(self.firstNameField, firstName)
        self.page.fill(self.lastNameField, lastName)
        self.page.fill(self.postalCodeField, postalCode)
        self.page.click(self.continueButton)

    def getCheckOutErrorMessage(self):
        return self.page.text_content(self.checkOutError)

    def finishOrder(self):
        self.page.click(self.finishButton)

    def getThankYouMessage(self):
        return self.page.text_content(self.thankYouMessage)

File: playwright-js-voidr/tests/pages/cart-page.js
// cartPage.js
const BasePage = require('./base-page');
const {
  firstNameField,
  lastNameField,
  postalCodeField,
  continueButton,
  finishButton,
  thankYouMessage,
  checkOutError,
  checkOutButton
} = require('../page-objects/cart-pages');

class CartPage extends BasePage {
  constructor(page) {
    super(page);
  }

  async checkout() {
    await this.page.click(checkOutButton);
  }

  async fillCheckOutInformation(firstName, lastName, postalCode) {
    await this.page.fill(firstNameField, firstName);
    await this.page.fill(lastNameField, lastName);
    await this.page.fill(postalCodeField, postalCode);
    await this.page.click(continueButton);
  }

  async getCheckOutErrorMessage() {
    return await this.page.textContent(checkOutError);
  }

  async finishOrder() {
    await this.page.click(finishButton);
  }

  async getThankYouMessage() {
    return await this.page.textContent(thankYouMessage);
  }
}

module.exports = CartPage;